service/history/historyEngine.go

```diff
@@ -2084,11 +2084,8 @@ func (e *historyEngineImpl) RequestCancelWorkflowExecution(
 			isCancelRequested, cancelRequestID := mutableState.IsCancelRequested()
 			if isCancelRequested {
 				cancelRequest := req.CancelRequest
-				if cancelRequest.RequestID != nil {
-					requestID := *cancelRequest.RequestID
-					if requestID != "" && cancelRequestID == requestID {
-						return updateWorkflowWithNewDecision, nil
-					}
+				if cancelRequest.RequestID != "" && cancelRequest.RequestID == cancelRequestID {
```

Comment:
This is the only place that has `!= nil` logic attached. However this can be simplified since `nil` will be mapped to empty string in mapping layer.

Reply:
I'm guessing `""` isn't a "valid" requestID anyway?

Reply:
Generally RequestID can be empty. At least it is not validated against, nor it is set it in all client requests.
However in all cases it used via GetRequestID helper, which turns nil to empty string anyway. Only this place checks agains nil directly, but this can be changed to equivalent check I did.